### PR TITLE
Support FFDHE in TLS 1.2 and below. Better TLS 1.3 version support.

### PR DESCRIPTION
**Pull request by SparkiDev**
Add support for the fixed FFDHE curves to TLS 1.2. Same curves in TLS
1.3 already. On by default - no checking of prime required.
Add option to require client to see FFDHE parameters from server as per
'may' requirements in RFC 7919.

Change TLS 1.3 ClientHello and ServerHello parsing to find the
SupportedVersions extension first and process it. Then it can handle
other extensions knowing which protocol we are using.

### CODE REVIEW

**Comment by dgarske** [comment]
@SparkiDev : Seems to be failing the Known Customer Config test with FAIL: tests/unit.test.

File: wolfssl/wolfcrypt/settings.h [hunk 1]
```diff
@@ -1808,6 +1808,37 @@ extern void uITRON4_free(void *p) ;
...
 #endif
 
+#if !defined(NO_DH) && !defined(HAVE_FFDHE)
+    #if defined(HAVE_FFDHE_2048) || defined(HAVE_FFDHE_3072) || \
+            defined(HAVE_FFDHE_4096) || defined(HAVE_FFDHE_6144) || \
+            defined(HAVE_FFDHE_8192)
+        #define HAVE_FFDHE
+    #endif
+#endif
+#ifdef FP_MAX_BITS
```
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: Should we detect whether this was user set explicitly versus us setting as a default.  Say a user only has so much memory and has set FP_MAX_BITS to handle that size.  Should we be increasing the size w/o a warning or detection?

Comment (SparkiDev) [reply wolfssl/wolfcrypt/settings.h]: Not sure what problem I was trying to solve here.
Changed to be a compiler error.
I don't want the user to compile in parameters that will cause random overruns and random failures.

Comment (SparkiDev) [reply wolfssl/wolfcrypt/settings.h]: Found the configuration that doesn't work!
It is a customer configuration - WOLFSSL_X86_64_BUILD set FP_MAX_BITS.
Fixing that now.

Comment (SparkiDev) [reply wolfssl/wolfcrypt/settings.h]: Fixed with latest push

**Pull request merged**